src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -183,6 +183,9 @@ function ($m) {
 
         restore_error_handler();
 
+        // Remove NULL characters to prevent DOMDocument::loadHTML() from stopping the document parsing
+        $content = str_replace("\x00", '', $content);
```

Comment:
This changes the content and effectively "concatenates" both sides of the `\00`, potentially changing the semantic of the document. Some XSS injections rely exactly on this kind of replacements...